canard.h

```diff
@@ -417,8 +429,10 @@ void canardPopTxQueue(CanardInstance* ins);
 /**
  * Processes a received CAN frame with a timestamp.
  * The application will call this function when it receives a new frame from the CAN bus.
+ *
+ * Return value will report any errors in decoding packets.
  */
-void canardHandleRxFrame(CanardInstance* ins,
+CanardError canardHandleRxFrame(CanardInstance* ins,
```

Comment:
`CanardError` --> `int16_t`

Reply:
This is an interesting point - what is the benefit of returning an `int16_t` over a `CanardError`?

There is a point that comes up with enumerations sometimes - they are always `int` type and so we can save memory by specifying a type.

I think in most applications the point is invalid - if we look at generated code an `int16_t` return value and a `enum` will both use a register and be the same size.

When we do things like:

```
void my_function(...)
{
    .... some variables
    `int16_t error`
    ....
```

The compiler will (almost..) certainly use a register for this error value - if the value isn't elided all together by the optimiser.

The only case I know where an `int` (and therefore `enum`) is bigger than a register is on PIC18 where `int` is 16 bits but registers are 8 bits - in this case using `int16_t` instead of `enum` will not make any difference.

A soft benefit of returning `CanardError` is that anyone can tell purely from the function prototype that the values the function returns will be from this enum.

Reply:
I'm happy to discuss this here - also happy if you would prefer to wait until I bring you a new PR.

Reply:
The reasons have nothing to do with optimization. There are functions that may return a meaningful non-negative value or an error, in which case the error code is negated. This approach is widely used in C, e.g. the Linux kernel. Since the returned value can be either an integral value or an error, you can't use enumeration there.